pkg/gardenlet/controller/shootstate/secret/reconciler.go

```diff
@@ -122,8 +123,13 @@ func (r *Reconciler) reconcile(
 		Type:   "secret",
 		Data:   runtime.RawExtension{Raw: dataJSON},
 	})
-	shootState.Spec.Gardener = dataList
 
+	// If the data list does not change, do not even try to send an empty PATCH request.
+	if apiequality.Semantic.DeepEqual(shootState.Spec.Gardener, dataList) {
+		return reconcile.Result{}, nil
+	}
+
```

Comment:
Didn't we still observe empty patches being sent after this check? 👀

Reply:
You're right, I double-checked again and adapted this check to ensure that it works as expected.